LibreHardwareMonitorLib/Hardware/Gpu/IntelGpuGroup.cs

```diff
@@ -55,7 +57,7 @@ public IntelGpuGroup(CPU.IntelCpu intelCpu, ISettings settings)
 
                             if (deviceInfo.Integrated)
                             {
-                                _hardware.Add(new IntelIntegratedGpu(intelCpu, deviceId, deviceInfo, settings));
+                                _hardware.Add(new IntelIntegratedGpu(intelCpus.First(), deviceId, deviceInfo, settings));
```

Comment:
It should create a new iGPU class for each CPU, otherwise it has the same result as the previous code :)

Reply:
added a comment to describe why this is like it is

Reply:
I don't think they exists (yet?), but I don't think this comment is correct. I know it's only used for the energy multiplier, but it might be different per CPU (although it seems unlikely that one would use different CPUs at the same time). I think the optimal solution would be to find a match between the GPU and CPU. But I'm not sure if there's any mapping between these two? I don't have an Intel CPU here with an iGPU.

Reply:
The way I understand it having different energy multipliers would mean you have CPUs from different micro-architectures, which I don't see happening. I do agree that matching the iGPU with the CPU would be more correct, in a multi CPU system we would want to do this to make sure we read the energy register on the correct CPU.

Matching the iGPU with the CPU could probably be done based on pci bus addresses, but it's really hard for me to write that code without having hardware like that to test.

I have also started looking into a more native approach to this using Intels oneAPI, but while I can connect to the iGPU, I have not had any luck reading any meaningful metrics from it. There is also a newer API (cotrollib.dll) included with the intel gpu drivers, but that requires a 12gen CPU or later which I don't have. I assume these two methods will also work on any future intel discrete GPUs (ARC). I'll experiment some more with these APIs later and make a followup PR if I have luck.

Reply:
Thanks, merged!